network-api/networkapi/buyersguide/management/commands/generate_pni_report.py

```diff
@@ -0,0 +1,84 @@
+import psycopg2
+from urllib.parse import urlparse
+from django.core.management.base import BaseCommand
+from django.conf import settings
+
+from networkapi.buyersguide.models import Product
+
+
+class Command(BaseCommand):
+    help = 'Generate statistical data about privacynotincluded activity, for internal' \
+           ' dashboards and update a DB connected to data studio'
+
+    def setup_db_connection(self):
+        STATS_DB = urlparse(settings.PNI_STATS_DB_URL)
+        return psycopg2.connect(
+            host=STATS_DB.hostname,
+            port=STATS_DB.port,
+            user=STATS_DB.username,
+            password=STATS_DB.password,
+            database=STATS_DB.path[1:]
+        )
+
+    def fetch_stats(self):
+        stats = []
+        for product in Product.objects.all():
+            votes = product.votes
+
+            creepiness_votes = 0
+            for bucket in votes['creepiness']['vote_breakdown']:
+                creepiness_votes += votes['creepiness']['vote_breakdown'][bucket]
+
+            stats.append({
+                'id': product.id,
+                'product': product.name,
+                'creepiness': votes['creepiness']['average'],
+                'creepiness_votes': creepiness_votes,
+                'would_buy': votes['confidence']['1'],
+                'would_not_buy': votes['confidence']['0']
+            })
+
+        return stats
+
+    def generate_insert_values(self, stats):
+        retval = ''
+        for index, p_data in enumerate(stats):
```

Comment:
this feels like it should be list comprehension with some glue?
```
return ', '.join([formstring(p_data) for p_data in enumerate(stats)])
```
with
```
def formstring(data):
  return "({id}, '{name}', {value}, {votes}, {yes}, {no})".format(
    id=data['id'],
    name=data['product'],
    value=data['creepiness'],
    votes=data['creepiness_votes'],
    yes=data['would_buy'],
    no=data['would_not_buy']
  )
```
(f strings are super great for autoname resolution, not so great for complex formatting like this)